### ISSUE TITLE
'policy' key empty in ReplayBuffer after collecting an episode

### ISSUE THREAD

**Issue by drozzy** [aai-institute/tianshou#476]
A bit confused by the key `policy` in the ReplayBuffer - [in the api docs](https://tianshou.readthedocs.io/en/master/api/tianshou.policy.html) it says that the _policy_ `can be accessed in the policy update process (e.g. in “policy.learn()”, the “batch.policy”`.

However, in the [documentation here](https://tianshou.readthedocs.io/en/master/tutorials/concepts.html#buffer) it seems like `policy` is part of the ReplayBuffer.

I thought that perhaps if I collect an episode - the _logits_ would should up under the `policy` key - but they are not there:

```python
from tianshou.data import ReplayBuffer
from tianshou.policy.modelfree.pg import PGPolicy
from tianshou.utils.net.common import Net
from tianshou.data.collector import Collector
import torch
import gym
import numpy as np

env = gym.make('CartPole-v0')
state_shape = env.observation_space.shape or env.observation_space.n
action_shape = env.action_space.shape or env.action_space.n

model = Net(state_shape,  action_shape, hidden_sizes=[16])
optim = torch.optim.Adam(model.parameters())
def dist(logits):
    return torch.distributions.Categorical(logits=logits)
policy = PGPolicy(model, optim, dist)

rb = ReplayBuffer(size=7)
collector = Collector(policy, env, rb)
collector.collect(n_episode=2)

print(rb.policy)
```

and the result I get is:
```
Batch()
```

Basically, I've no idea what `policy` key does in ReplayBuffer while trying to explain it to someone.

**Comment by Trinkle23897** [comment]
`policy` is for storing intermediate result in `policy.forward` function. If the returned Batch has a key `policy`, e.g. `return Batch(policy=logits, ...)`, this type of data will stored into the replay buffer.

But currently none of policy classes use this feature, so that you'll see the policy is an empty batch.

**Comment by drozzy** [comment]
Ok, great thanks. If anyone in the future needs an example of this, I made this snippet:

```python

from tianshou.data import ReplayBuffer, Batch
from tianshou.policy.modelfree.pg import PGPolicy
from tianshou.utils.net.common import Net
from tianshou.data.collector import Collector
import torch
import gym

class MyPolicy(PGPolicy):
    def forward(self, *args, **kwargs):
        batch = super().forward(*args, **kwargs)
        batch['policy'] = Batch({'logits': batch.logits})
        
        return batch

env = gym.make('CartPole-v0')
state_shape = env.observation_space.shape or env.observation_space.n
action_shape = env.action_space.shape or env.action_space.n

model = Net(state_shape,  action_shape, hidden_sizes=[16])
optim = torch.optim.Adam(model.parameters())
def dist(logits):
    return torch.distributions.Categorical(logits=logits)
policy = MyPolicy(model, optim, dist)

rb = ReplayBuffer(size=7)
collector = Collector(policy, env, rb)

collector.collect(n_episode=2)
print(rb)
```

Output will be:
```python
ReplayBuffer(
    rew: array([1., 1., 1., 1., 1., 1., 1.]),
    done: array([False, False,  True, False, False, False, False]),
    act: array([1, 0, 0, 0, 1, 0, 0]),
    policy: Batch(
                logits: tensor([[ 0.1042, -0.2839],
                                [ 0.0904, -0.2617],
                                [ 0.1198, -0.3014],
                                [ 0.0637, -0.1933],
                                [ 0.0673, -0.2373],
                                [ 0.0668, -0.2037],
                                [ 0.0765, -0.2473]]),
            ),
    obs_next: array([[-0.18010915, -0.82770361,  0.15723808,  1.24394865],
                     [-0.19666323, -1.02445419,  0.18211705,  1.58147146],
                     [-0.21715231, -1.22121642,  0.21374648,  1.9249808 ],
                     [-0.11416554, -0.82278793,  0.06440773,  1.12404367],
                     [-0.1306213 , -0.62856666,  0.08688861,  0.85223862],
                     [-0.14319263, -0.82475886,  0.10393338,  1.17092986],
                     [-0.15968781, -1.02106728,  0.12735198,  1.49430493]]),
    obs: array([[-0.15968781, -1.02106728,  0.12735198,  1.49430493],
                [-0.18010915, -0.82770361,  0.15723808,  1.24394865],
                [-0.19666323, -1.02445419,  0.18211705,  1.58147146],
                [-0.1016247 , -0.62704192,  0.04807503,  0.81663529],
                [-0.11416554, -0.82278793,  0.06440773,  1.12404367],
                [-0.1306213 , -0.62856666,  0.08688861,  0.85223862],
                [-0.14319263, -0.82475886,  0.10393338,  1.17092986]]),
    info: Batch(
              env_id: array([0, 0, 0, 0, 0, 0, 0]),
          ),
)
```
